lib/Driver/ToolChains/Hcc.cpp

```diff
@@ -137,74 +153,151 @@ void HCC::CXXAMPAssemble::ConstructJob(Compilation &C, const JobAction &JA,
   C.addCommand(llvm::make_unique<Command>(JA, *this, Exec, CmdArgs, Inputs));
 }
 
-std::string temporaryReplaceLongFormGFXIp(const Compilation& C, std::string l)
-{ // Precondition: l = "AMD:AMDGPU:\d:\d:\d"
-  // TODO: this should be removed once we have transitioned all users to using
-  //       the short form. It is purposefully inefficient.
-  const auto t = l;
+namespace
+{
+    std::string temporaryReplaceLongFormGFXIp(
+        const Compilation& C, std::string l)
+    { // Precondition: l = "AMD:AMDGPU:\d:\d:\d"
+        // TODO: this should be removed once we have transitioned all users to
+        //       the short form. It is purposefully inefficient.
+        const auto t = l;
+
+        l.replace(0u, 3u, {'g', 'f', 'x'});
+        l.erase(std::copy_if(
+            l.begin() + 3u, l.end(), l.begin() + 3u, isdigit), l.end());
+        C.getDriver().Diag(diag::warn_drv_deprecated_arg) << t << l;
+
+        return l;
+    }
+
+    std::pair<std::vector<std::string>, int> detect_gfxip(
+        const Compilation& C, const ToolChain& TC)
+    {
+        std::vector<std::string> agents;
+
+        const auto e = C.getSysRoot() + "/opt/rocm/bin/rocm_agent_enumerator";
```

Comment:
We need to find a way to make this path configurable.  (i.e. get the value of ROCM_ROOT in CMakefile here)